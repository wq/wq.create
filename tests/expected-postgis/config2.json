{
    "pages": {
        "login": {
            "url": "login",
            "name": "login"
        },
        "logout": {
            "url": "logout",
            "name": "logout"
        },
        "location": {
            "cache": "first_page",
            "background_sync": true,
            "map": [
                {
                    "mode": "list",
                    "layers": [
                        {
                            "name": "Locations",
                            "type": "geojson",
                            "data": [
                                "context_feature_collection",
                                "geometry"
                            ],
                            "popup": "location",
                            "cluster": true
                        },
                        {
                            "name": "All Locations",
                            "type": "vector-tile",
                            "layer": "location",
                            "popup": "location",
                            "identifyLayers": [
                                "location-circle"
                            ],
                            "active": false
                        }
                    ]
                },
                {
                    "mode": "detail",
                    "layers": [
                        {
                            "name": "Location",
                            "type": "geojson",
                            "data": [
                                "context_feature",
                                "geometry"
                            ],
                            "popup": "location"
                        },
                        {
                            "name": "All Locations",
                            "type": "vector-tile",
                            "layer": "location",
                            "popup": "location",
                            "identifyLayers": [
                                "location-circle"
                            ],
                            "active": false
                        }
                    ]
                },
                {
                    "mode": "edit",
                    "layers": [
                        {
                            "name": "All Locations",
                            "type": "vector-tile",
                            "layer": "location",
                            "popup": "location",
                            "active": false
                        }
                    ]
                }
            ],
            "name": "location",
            "url": "locations",
            "list": true,
            "form": [
                {
                    "name": "name",
                    "label": "Site Name",
                    "bind": {
                        "required": true
                    },
                    "hint": "The unique name of the site",
                    "type": "text"
                },
                {
                    "name": "type",
                    "label": "Site Type",
                    "bind": {
                        "required": true
                    },
                    "hint": "Site classification",
                    "choices": [
                        {
                            "name": "water",
                            "label": "Water"
                        },
                        {
                            "name": "air",
                            "label": "Air"
                        },
                        {
                            "name": "land",
                            "label": "Land"
                        }
                    ],
                    "type": "select one"
                },
                {
                    "name": "geometry",
                    "label": "Location",
                    "bind": {
                        "required": true
                    },
                    "hint": "The location of the site",
                    "type": "geopoint"
                }
            ],
            "verbose_name": "location",
            "verbose_name_plural": "locations",
            "geometry_fields": [
                {
                    "name": "geometry",
                    "label": "Location",
                    "type": "geopoint"
                }
            ]
        },
        "index": {
            "url": "",
            "name": "index",
            "show_in_index": false,
            "verbose_name": "test Project",
            "map": {
                "mapId": "map",
                "layers": [
                    {
                        "name": "Locations",
                        "type": "vector-tile",
                        "layer": "location",
                        "popup": "location",
                        "identifyLayers": [
                            "location-circle"
                        ]
                    }
                ]
            }
        }
    },
    "site_title": "test Project",
    "router": {
        "base_url": ""
    },
    "store": {
        "service": "",
        "defaults": {
            "format": "json"
        }
    },
    "logo": "/static/app/images/icon-192.png",
    "material": {
        "theme": {
            "primary": "#7500ae",
            "secondary": "#0088bd"
        }
    },
    "map": {
        "bounds": [
            [
                -180,
                -70
            ],
            [
                180,
                70
            ]
        ],
        "tiles": "/tiles/{z}/{x}/{y}.pbf"
    },
    "debug": true
}
