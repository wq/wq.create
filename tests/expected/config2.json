{
    "pages": {
        "login": {
            "url": "login",
            "name": "login"
        },
        "logout": {
            "url": "logout",
            "name": "logout"
        },
        "location": {
            "cache": "first_page",
            "background_sync": true,
            "map": [
                {
                    "mode": "list",
                    "autoLayers": true,
                    "layers": []
                },
                {
                    "mode": "detail",
                    "autoLayers": true,
                    "layers": []
                },
                {
                    "mode": "edit",
                    "autoLayers": true,
                    "layers": []
                }
            ],
            "name": "location",
            "url": "locations",
            "list": true,
            "form": [
                {
                    "name": "name",
                    "label": "Site Name",
                    "bind": {
                        "required": true
                    },
                    "hint": "The unique name of the site",
                    "type": "text"
                },
                {
                    "name": "type",
                    "label": "Site Type",
                    "bind": {
                        "required": true
                    },
                    "hint": "Site classification",
                    "choices": [
                        {
                            "name": "water",
                            "label": "Water"
                        },
                        {
                            "name": "air",
                            "label": "Air"
                        },
                        {
                            "name": "land",
                            "label": "Land"
                        }
                    ],
                    "type": "select one"
                },
                {
                    "name": "geometry",
                    "label": "Location",
                    "bind": {
                        "required": true
                    },
                    "hint": "The location of the site",
                    "type": "geopoint"
                }
            ],
            "verbose_name": "location",
            "verbose_name_plural": "locations"
        },
        "index": {
            "url": "",
            "name": "index",
            "show_in_index": false,
            "verbose_name": "test Project"
        }
    },
    "site_title": "test Project",
    "router": {
        "base_url": ""
    },
    "store": {
        "service": "",
        "defaults": {
            "format": "json"
        }
    },
    "material": {
        "theme": {
            "primary": "#7500ae",
            "secondary": "#0088bd"
        }
    },
    "map": {
        "bounds": [
            [
                -180,
                -70
            ],
            [
                180,
                70
            ]
        ]
    },
    "debug": true
}
